### ISSUE TITLE
bug(no-bare-urls): standalone URL on its own line should not be flagged

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#173]
## Overview

`no-bare-urls` incorrectly flags a bare URL that stands alone on its own line surrounded by blank lines. This pattern is intentional — many Markdown renderers (GitHub, Zenn, etc.) treat it as a **link card** embed, not an unstyled bare URL in prose.

## Problem

A URL written as a standalone paragraph:

```markdown
Some text above.

https://example.com

Some text below.
```

is currently reported as a violation:

```
no-bare-urls: bare URL found, use angle brackets or a Markdown link: https://example.com
```

This is a false positive. The author deliberately placed the URL alone to trigger a link card preview. Requiring them to wrap it in `<>` or `[text](url)` would break the intended rendering.

## Expected behavior

When a line contains **only** a bare URL (no surrounding prose) and is preceded and followed by a blank line (or file boundary), `no-bare-urls` should treat it as a link card and skip it.

## Detection logic

| Pattern | Should flag? |
|---|---|
| `Visit https://example.com for details.` | Yes — URL embedded in prose |
| `(https://example.com)` | Yes — URL in parentheses within prose |
| `https://example.com` alone on a line, surrounded by blank lines | **No** — link card intent |
| `https://example.com` on the first line of file with blank line after | **No** — link card intent |
| `https://example.com` on the last line of file with blank line before | **No** — link card intent |
| `https://example.com https://other.com` (two URLs on one line) | Yes — not a clean standalone link card |

## Fix

In `CheckNoBareURLs`, before scanning a line for bare URLs, check whether:

1. The trimmed line equals a single bare URL (the full line is just the URL), **and**
2. The previous line is blank (or it is the first line), **and**
3. The next line is blank (or it is the last line).

If all three conditions hold, skip the line.

## References

- [GitHub link cards](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams)
- [Zenn link card](https://zenn.dev/zenn/articles/markdown-guide#%E3%83%AA%E3%83%B3%E3%82%AF%E3%82%AB%E3%83%BC%E3%83%89)
- Rule source: `internal/rule/no_bare_urls.go`

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

- `internal/rule/no_bare_urls_test.go` — add test cases for standalone URL (already exists; add cases)

### Files to modify

- `internal/rule/no_bare_urls.go` — add `isStandaloneURL(lines []string, i int)` helper and skip logic in `CheckNoBareURLs`

### Checklist

- [ ] Add `isStandaloneURL` helper: checks line `i` is a single bare URL with blank lines on both sides (or file boundary)
- [ ] Add skip in `CheckNoBareURLs` when `isStandaloneURL` returns true
- [ ] Add test cases: standalone URL at start / middle / end of file; two URLs on one line; URL with trailing prose
- [ ] `make test`
- [ ] `make test-e2e`
- [ ] `make test-all`
